### PR TITLE
Resequencing haplotype vcf

### PR DESCRIPTION
**Pull request by lynnjo**
## Description

Adding commands to create a haplotype vcf from imputation and variant calling results.

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [X] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Added new commands composite-to-haplotype-coords and imputation-metrics
```

### CODE REVIEW

File: src/main/kotlin/net/maizegenetics/phgv2/cli/CreateHaplotypeVCF.kt [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+ * The incoming variantVCF file is in composite genome coordinates.
+ * The incoming path hvcf file is in reference coordinates.
+ * The output file will be in haplotype coordinates, calculated from the Regions information
+ * in the path hvcf file's ALT header information, combined with the reference coordinates from the path hvcf entries.
+ *
+ *
+ */
+data class HaplotypeData(val id: String, val refContig: String, val refStart: Int, val hapLen: Int)
+
+class CreateHaplotypeVCF: CliktCommand(help = "Create g.vcf file for a PHG pathing h.vcf using data from existing PHG created g.vcf files")  {
```
Comment (zrm22) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/CreateHaplotypeVCF.kt:37]: I wonder if this needs to be named something else.  At first glance it seems that it just makes a hvcf but doesn't really say for what.  Maybe CreateRareAlleleVCF or CreateReseqeunceVCF?

File: src/main/kotlin/net/maizegenetics/phgv2/pathing/ImputationMetrics.kt [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ * or in the parent sample h.vcf file, and a column that indicates if the 2 haplotypes (parent
+ * and imputation) are the same.
+ *
+ * The output file will be tab-delimited and will be written to the output file specified
+ * by the user.
+ */
+
+class ImputationMetrics : CliktCommand(help = "Impute best path(s) using read mappings.")   {
+    private val myLogger = LogManager.getLogger(ImputationMetrics::class.java)
+    val sampleHvcf by option(help = "Path to hvcf file loaded to tiledb for the Parent sample of your short reads. For example: If your reads are from P39, this would be the P39.h.vcf.Required parameter.")
```
Comment (zrm22) [inline src/main/kotlin/net/maizegenetics/phgv2/pathing/ImputationMetrics.kt:35]: With Terry's new changes does the Required parameter in the help need to be there?  Or does the help already show it?

**Review by zrm22 — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/pathing/ImputationMetrics.kt (hunk 2 above)
Comment (tcasstevens) [reply src/main/kotlin/net/maizegenetics/phgv2/pathing/ImputationMetrics.kt:35]: No it will say required automatically

**Review by tcasstevens — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/CreateHaplotypeVCF.kt (hunk 1 above)
Comment (aberthel) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateHaplotypeVCF.kt:37]: Maybe something about coordinates, like CreateHapCoordVCF? Still a little long, but it gets across that the coordinates are being manipulated.

**Review by aberthel — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/pathing/ImputationMetrics.kt (hunk 2 above)
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/pathing/ImputationMetrics.kt:35]: To clarify - you would like me to remove "Required parameter" from each of the inline parameter descriptions?

Comment (zrm22) [reply src/main/kotlin/net/maizegenetics/phgv2/pathing/ImputationMetrics.kt:35]: yes I think the 'Required parameter." portion of the help is not needed anymore.

File: src/main/kotlin/net/maizegenetics/phgv2/cli/CreateHaplotypeVCF.kt (hunk 1 above)
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateHaplotypeVCF.kt]: As per lab meeting today, Ed says to pick something and we can change later.  I have picked composite-to-haplotype-coords based on Cinta, Bethany and others comments.  I do not expect everyone to like this.  Will let Brandon make changes when/if he updates documentation.

**Pull request merged**